Added model parameter to OpenAIEmbeddings, enable AzureOpenAI

ghaccount opened the pull request:
consistent with openai package. AzureOpenAI can be used by setting env vars. Updated doc as well.

ghaccount committed:
Added model parameter to OpenAIEmbeddings. This enables AzureOpenAI as well, consistent with openai package. AzureOpenAI can be used by setting of env vars. Updated doc as well.

ghaccount committed:
Merge branch 'hwchase17:master' into master

hwchase17 reviewed (commented):
why did you end up deleting the document model name and query model name? 

other than that looks good

ghaccount committed:
Added back support for first generation models with separate query_model_name/document_model_name. The second generation is recommended -https://openai.com/blog/new-and-improved-embedding-model

Added the example of using model_name param for first generation models.
For second generation, with single model, model param should be used, works with Azure as well with model=deployment_name

ghaccount committed:
Merge branch 'master' of https://github.com/ghaccount/langchain

ghaccount committed:
Merge branch 'hwchase17:master' into master

ghaccount commented:
@hwchase17 Updated the PR with document model name and query model name. I was proposing to simplify removing those two - https://github.com/openai/openai-python/issues/132. Perhaps you can plan that in future and move to model param. I mentioned model param for 2nd gen models. While at it, also added doc for usage first gen for model_name one current supported.

ghaccount committed:
fix the docstring

ghaccount committed:
Merge branch 'hwchase17:master' into master

ghaccount committed:
Merge branch 'hwchase17:master' into master
